polynote-kernel/src/main/scala/polynote/config/PolynoteConfig.scala

```diff
@@ -163,10 +162,20 @@ object Credentials {
   implicit val decoder: Decoder[Credentials] = deriveDecoder
 }
 
+final case class ScalaVersionConfig(
+  versionNumber: String,
+  sparkSubmitArgs: String
+)
+
+object ScalaVersionConfig {
+  implicit val encoder: Encoder.AsObject[ScalaVersionConfig] = deriveEncoder
+  implicit val decoder: Decoder[ScalaVersionConfig] = deriveDecoder
+}
+
 final case class SparkPropertySet(
   name: String,
   properties: ShortMap[String, String] = ShortMap(Map.empty[String, String]),
-  sparkSubmitArgs: Option[String] = None,
+  versionConfigs: Option[List[ScalaVersionConfig]] = None,
```

Comment:
I think it would be preferable to make this change backwards-compatible if at all possible. We don't want to break configs that are out there in the wild

Reply:
Gotcha! https://github.com/polynote/polynote/pull/1432/files#diff-7745c204a4ffba846be57054277a6f621c214efa21c0d706eb4dfc66c700885cR189-R191